From 6d9c390dcb83ca05d3c05919a0d363c41292545c Mon Sep 17 00:00:00 2001
From: Michalis Pappas <Michalis.Pappas@opensynergy.com>
Date: Fri, 19 Nov 2021 18:48:24 +0100
Subject: [PATCH] Disable calls to exit() on core, utils2, utils3

In Unikraft all tests are executing in the scope of a single
application, and calling exit() causes the premature termination of the
application. Disable exit() on core, utils2, and utils3.
---
 test/default/sodium_core_test.c   | 2 ++
 test/default/sodium_utils2_test.c | 3 +--
 test/default/sodium_utils3_test.c | 3 +--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/test/default/sodium_core_test.c b/test/default/sodium_core_test.c
index f0ce22c..e22aac6 100644
--- a/test/default/sodium_core_test.c
+++ b/test/default/sodium_core_test.c
@@ -2,6 +2,8 @@
 #define TEST_NAME "sodium_core"
 #include "cmptest.h"
 
+#define __EMSCRIPTEN__
+
 static void
 misuse_handler(void)
 {
diff --git a/test/default/sodium_utils2_test.c b/test/default/sodium_utils2_test.c
index 80c3451..a797770 100644
--- a/test/default/sodium_utils2_test.c
+++ b/test/default/sodium_utils2_test.c
@@ -16,7 +16,7 @@
 #undef sodium_free
 #undef sodium_allocarray
 
-__attribute__((noreturn)) static void
+static void
 segv_handler(int sig)
 {
     (void) sig;
@@ -34,7 +34,6 @@ segv_handler(int sig)
     signal(SIGABRT, SIG_DFL);
 # endif
 #endif
-    exit(0);
 }
 
 int
diff --git a/test/default/sodium_utils3_test.c b/test/default/sodium_utils3_test.c
index 405c4a0..b134fd3 100644
--- a/test/default/sodium_utils3_test.c
+++ b/test/default/sodium_utils3_test.c
@@ -12,7 +12,7 @@
 # warning The sodium_utils3 test is expected to fail with address sanitizer
 #endif
 
-__attribute__((noreturn)) static void
+static void
 segv_handler(int sig)
 {
     (void) sig;
@@ -30,7 +30,6 @@ segv_handler(int sig)
     signal(SIGABRT, SIG_DFL);
 # endif
 #endif
-    exit(0);
 }
 
 int
-- 
2.25.1

